### PR TITLE
Initial README for core

### PR DESCRIPTION
**Pull request by bbuchalter**
Intended to provide a high-level over of the features of core. This is a work in progress and is not complete, but hopefully provides a format and some inspiration to do this important work! Would welcome PRs to my branch to move this past WIP.

Available at https://github.com/bbuchalter/solidus/blob/core-readme/core/README.md for visual review.

### CODE REVIEW

**Comment by bbuchalter** [comment]
Although I thought including a summary of each model would be helpful, there are so many, it's likely going to overwhelm the reader. I do like breaking things into sub-systems and providing a very terse summary of each model. Perhaps these longer summaries could go into docs in the model files themselves? Bonus points for having this README auto generated from the docs so it doesn't go stale overtime.

**Comment by jrochkind** [comment]
Nice work!

**Comment by bbuchalter** [comment]
Style question: should these terse descriptions reference specific classes or the concepts which a class represents? For example, which of these is preferred?

`Spree::ReturnAuthorization` - Models the return of Inventory Units to a Stock Location for an Order.

or

`Spree::ReturnAuthorization` - Models the return of `Spree::InventoryUnit`s to a `Spree::StockLocation` for a `Spree::Order`.

**Comment by bbuchalter** [comment]
I was thinking it would be nice to add high-level summaries of each sub-system. I was also thinking this may be an appropriate place to link to the old Spree Guides, with a warning that they may conflict with the current state of Solidus, but generally should represent a reasonable introduction. Thoughts?

**Comment by bbuchalter** [comment]
> I was also thinking this may be an appropriate place to link to the old Spree Guides

I guess this was discussed in https://github.com/solidusio/solidus/issues/852 which lead to https://github.com/solidusio/solidus_guides but sadly, there has been very little activity there and the output from that repo (http://ashkamel.com/solidus-guides/users/orders/) doesn't seem ready for prime time.

**Comment by Kingdutch** [comment]
Just wanted to chime in that the work you've done so far looks really good : ) 
Gives a good overview of the way Solidus approaches solving the e-commerce challenge. 

I also like the comment about the Spree namespacing as that was something that first tripped me up too.

As a new developer it gives me some nice handles to start exploring. I realise there's the normal rails structure, but that makes 200 models still a bit overwhelming. With this set-up I can see what the most important models are and then start exploring from there, finding the dependencies and helper classes.

**Comment by bbuchalter** [comment]
@Kingdutch thanks for the encouragement. It's going to take a while to make this comprehensive enough, but I figure I can chip away at it over time.

File: core/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+and thus a deep inspection of each in this README would be overkill. Instead,
+let's take a quick look at the fundamental models upon which all else depend.
+Currently, these models remain in the Spree namespace as part of the legacy of
+[forking Spree](https://solidus.io/blog/2015/10/28/future-of-spree.html).
+
+* [The Order Sub-System](https://github.com/bbuchalter/solidus/blob/core-readme/core/README.md#the-order-sub-system)
+* [The User Sub-System](https://github.com/bbuchalter/solidus/blob/core-readme/core/README.md#the-user-sub-system)
+* [The Payment Sub-System](https://github.com/bbuchalter/solidus/blob/core-readme/core/README.md#the-payment-sub-system)
+* [The Inventory Sub-System](https://github.com/bbuchalter/solidus/blob/core-readme/core/README.md#the-inventory-sub-system)
+
```
Comment (isaacfreeman) [inline core/README.md]: I'm not sure if this list is intended to be complete yet, but I'd suggest adding

- Product (including products, variants, taxonomies, properties, options and prices)
- Promotion

File: core/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Solidus implements over 200 [models](https://github.com/solidusio/solidus/tree/master/core/app/models/spree),
+and thus a deep inspection of each in this README would be overkill. Instead,
+let's take a quick look at the fundamental models upon which all else depend.
+Currently, these models remain in the Spree namespace as part of the legacy of
+[forking Spree](https://solidus.io/blog/2015/10/28/future-of-spree.html).
+
+* [The Order Sub-System](https://github.com/bbuchalter/solidus/blob/core-readme/core/README.md#the-order-sub-system)
+* [The User Sub-System](https://github.com/bbuchalter/solidus/blob/core-readme/core/README.md#the-user-sub-system)
+* [The Payment Sub-System](https://github.com/bbuchalter/solidus/blob/core-readme/core/README.md#the-payment-sub-system)
+* [The Inventory Sub-System](https://github.com/bbuchalter/solidus/blob/core-readme/core/README.md#the-inventory-sub-system)
```
Comment (isaacfreeman) [inline core/README.md]: "Stock" might be a better term than "Inventory", as it's what's used in the backend menu.

Comment (bbuchalter) [inline core/README.md]: I don't have strong opinions on this either way. Anyone else want ot chime in?

**Comment by cbrunsdon** [comment]
All this looks reasonable to me, all better than what we have now, so :+1:

**Comment by bbuchalter** [comment]
Given the thumbs we've got for things as-is, I'm removing the WIP from this PR so it can be merged with additional documentation to be added in this form in the future. I've rebased over master and added a note indicating the documentation is still in progress and contributions are welcome. Please let me know what else is required to get this merged.

**Comment by jhawthorn** [comment]
Thanks so much Brian.

**Pull request merged**